### ISSUE TITLE
twisted.python.test.test_util should be ported to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7940]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7940|
|Type|enhancement|
|Created|2015-06-14 14:14:11Z|
|Branch|https://github.com/twisted/twisted/tree/tp-testutil-7940|

There's a lot of things in here which have dependencies already ported.

<details><summary>Searchable metadata</summary>

```
trac-id__7940 7940
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_tp_testutil_7940 branches/tp-testutil-7940
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1434291251706925 1434291251706925
changetime__1435510809276364 1435510809276364
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [45044]) Branching to tp-testutil-7940.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Builds are looking green, 400 less uncovered lines -- ```test_util``` is at 92% on Python 3.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Changes look good. Many thanks for working on trial.

Don't forget to run the test on latest revision.

-------

For

```

skip = "Python 3 has its own OrderedDict implementation."
```

Maybe we should enable those test on py3... in this way we make sure that py2 and py3 implementation provide the same functionality ... verified using a single test suite.

----

Just a very minor comment... maybe we should port the new code focused on writing py3 which works on py2, rather than py2 which works on py3...

For example, instead of 

```
__next__ = next
```

Maybe we should rename the current implementation to __next__ and use

```
# Support for py2
next = __next__
```

In this way, the future developlemt focus would be done on the __next__ method. 

--------

Please consider enabling the OrderedDict test on Py3 and merge if all is ok.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Hi Adi, thanks for the review.

I've fixed those two things up and am now just waiting for the builders.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45124]) Merge tp-testutil-7940: twisted.python.test.test_util should be ported to Python 3

Author: hawkowl
Reviewer: adiroiban
Fixes: [#7940](https://github.com/twisted/twisted/issues/7940)
